File: Assets/Samples/ModuTree/JankenSample/Nodes/SelectHandAction.cs
using System.Threading;
using System.Threading.Tasks;
using ModuTree.Runtime.Core;
using ModuTree.Runtime.Nodes;

namespace ModuTree.JankenSample
{
    /// <summary>
    /// AIが出す手を Blackboard に書き込むアクション。
    /// hand フィールドで指定した手を aiHand キーにセットして Success を返す。
    /// </summary>
    [BehaviourNodeMeta(
        displayName: "手を選択",
        category:    "Action",
        description: "指定した hand を aiHand に書き込んで Success",
        color:       "#1E3A6E")]
    public class SelectHandAction : ActionNodeData
    {
        [NodeField("選ぶ手", "AIが出す手")]
        public JankenHand hand = JankenHand.Rock;

        protected override Task<NodeState> OnUpdateAsync(CancellationToken ct)
        {
            Blackboard.Set(JankenBBKeys.AiHand, hand);
            return Task.FromResult(NodeState.Success);
        }
    }
}

File: Assets/Samples/ModuTree/JankenSample/PHP/Nodes/SelectHandAction.php
<?php
declare(strict_types=1);

/**
 * AIが出す手を Blackboard に書き込むアクション。
 * hand フィールドで指定した手を aiHand キーにセットして Success を返す。
 *
 * JSONパラメータ例:
 *   {"hand": 2}  // JankenHand::Paper（パー）でグーに勝つ
 *   {"hand": 0}  // JankenHand::Rock（グー）でチョキに勝つ
 *   {"hand": 1}  // JankenHand::Scissors（チョキ）でパーに勝つ
 */
class SelectHandAction extends ActionNodeData
{
    /** AIが出す手。JSONから自動復元される（int→JankenHand変換）。 */
    public JankenHand $hand = JankenHand::Rock;

    protected function onUpdate(): NodeState
    {
        $this->blackboard->set(JankenBBKeys::$aiHand, $this->hand);
        return NodeState::Success;
    }
}